Added a pattern for homepage

Imran92 opened the pull request:
Implements https://github.com/Automattic/sensei-theme/issues/33

## Changes proposed

- Added a pattern with relevant font, template and styles for theme homepage.

## Testing instructions

- Create a new page
- In the editor, change the template of the page to **Page - No Title**
- Now clear if there's any block on the page and click on the inserter
- Click on patterns Tab
- Click on the dropdown for pattern categories and select "Sensei LMS" from there
- Make sure the **Landing Page** pattern is there, select it.
- Click on the pattern, save the page, and go to the frontend. Make sure the design matches

## Screenshots and Videos

**Desktop**

https://user-images.githubusercontent.com/6820724/196554308-2cae64a0-d273-43dc-8dcb-2c471386a8cc.mov

**Mobile**

https://user-images.githubusercontent.com/6820724/196554366-4cdb09f9-40d9-4260-91dd-3b971af23be0.mov

gabrielcaires commented on patterns/homepage.php:
```diff
@@ -0,0 +1,230 @@
+ */
+?>
+
+<!-- wp:group {"align":"full","style":{"spacing":{"blockGap":"0px","padding":{"top":"0px","right":"0px","bottom":"0px","left":"0px"}}},"className":"course-homepage-pattern-parent-group","layout":{"inherit":false,"wideSize":"","contentSize":""}} -->
```
I think we should split this pattern into small parts to be reused on the other screens.
My suggestion is
- Grow your writing skills with the course is a pattern
- Course list 
- Blog Post
-The ` Keep track of the latest news and lessons. Every week in your inbox.`
-Testimonial

And we should use the home pattern only to connect the small patterns.
We have a lot of screens using the same blocks so we should not rewrite the same instructions

WDT? 

Cc @donnapep

donnapep replied:
@Automattic/theam Any thoughts on this, or on this PR in general?

Imran92 replied:
Thanks for the suggestion @gabrielcaires , I've broken it into reusable smaller patterns here https://github.com/Automattic/sensei-theme/pull/44/commits/0bf443111747fd27d9bc2a24346d753de739cc03, can you take a look?

gabrielcaires commented on patterns/homepage.php:
```diff
@@ -0,0 +1,230 @@
+<!-- /wp:group -->
+
+<!-- wp:heading {"textAlign":"center","className":"has-sorts-mill-goudy-font-family","fontSize":"5x-large"} -->
+<h2 class="has-text-align-center has-sorts-mill-goudy-font-family has-5-x-large-font-size">COURSE</h2>
```
I think we should move the footer to the template part instead of keeping it here.

Imran92 replied:
Right, moved it here https://github.com/Automattic/sensei-theme/pull/44/commits/0f0b9e3c852b17e88f97b7e7269bea46067189cc

gabrielcaires commented on style.css:
```diff
@@ -227,3 +227,80 @@ body .is-layout-flow .wp-block-group {
+	border: 1px solid var(--wp--preset--color--tertiary);
+	border-radius: 4px;
+}
+}
```
is this  extra `}` correct?

scruffian replied:
Do you need this CSS or can you put the settings in the blocks? If the settings are inline in the block then its easier for users to change them...

Imran92 replied:
@gabrielcaires thanks for noticing, I've removed it here https://github.com/Automattic/sensei-theme/pull/44/commits/b87083c1c01f4c84fdced59affdfc65b028f13f9 :)

Imran92 replied:
@scruffian thanks! :) I've removed all the styles which we could apply inline from here in this update https://github.com/Automattic/sensei-theme/pull/44/commits/0ac98bb70032bc77b6b62697655cb1e851b378e1

gabrielcaires commented on theme.json:
```diff
@@ -228,6 +262,18 @@
+				"elements": {
+					"h2": {
+						"typography": {
+							"fontFamily": "var(--wp--preset--font-family--league-gothic)"
```
The h2 font family is already declared here https://github.com/Automattic/sensei-theme/blob/trunk/theme.json#L562.
So I think this line is unnecessary.

Imran92 replied:
Thanks @gabrielcaires , removed it here https://github.com/Automattic/sensei-theme/pull/44/commits/139ed3b508555de4fa89647526be8dc6edadfc81

gabrielcaires commented on patterns/homepage.php:
```diff
@@ -0,0 +1,230 @@
+
+<!-- wp:group {"className":"has-eb-garamond-font-family","layout":{"type":"flex","flexWrap":"nowrap"}} -->
+<div class="wp-block-group has-eb-garamond-font-family"><!-- wp:paragraph {"fontSize":"small"} -->
+<p class="has-small-font-size">About</p>
```
I think these links are actually page links, should we use a nav block?

Imran92 replied:
Thanks @gabrielcaires , I looked a bit into using nav block here. It seems nav block is more suitable for the navbar and it's mostly used in that context. I think maybe we can keep it this way for now

gabrielcaires commented on functions.php:
```diff
@@ -68,6 +68,47 @@ function course_theme_init() {
+	);
+
+	$block_pattern_categories = array(
+		'sensei-lms' => array( 'label' => __( 'Sensei LMS', 'course' ) )
```
Why do we need to translate `Sensei LMS` ?

Imran92 replied:
Removed it here https://github.com/Automattic/sensei-theme/pull/44/commits/73331765814cfc083618f8a463f16acd60d6cde1

gabrielcaires commented on functions.php:
```diff
@@ -68,6 +68,47 @@ function course_theme_init() {
+	 */
+	$block_pattern_categories = apply_filters( 'course_block_pattern_categories', $block_pattern_categories );
+
+	foreach ( $block_pattern_categories as $name => $properties ) {
```
Could you please provide more context about why we need to use PHP only to register the pattern?
it theoretically should be automatically, rigth?

Imran92 replied:
Thanks @gabrielcaires , it is update here https://github.com/Automattic/sensei-theme/pull/44/commits/73331765814cfc083618f8a463f16acd60d6cde1

gabrielcaires commented on patterns/homepage.php:
```diff
@@ -0,0 +1,230 @@
+<!-- wp:group {"backgroundColor":"primary","textColor":"tertiary","layout":{"type":"flex","flexWrap":"nowrap","justifyContent":"space-between","verticalAlignment":"center"}} -->
+<div class="wp-block-group has-tertiary-color has-primary-background-color has-text-color has-background"><!-- wp:group {"style":{"spacing":{"padding":{"left":"20px"}}},"borderColor":"tertiary","className":"is-style-group-left-border","layout":{"type":"flex","flexWrap":"nowrap","justifyContent":"left"}} -->
+<div class="wp-block-group is-style-group-left-border has-border-color has-tertiary-border-color" style="padding-left:20px"><!-- wp:heading {"level":3,"style":{"typography":{"textTransform":"uppercase"}},"className":"has-league-gothic-font-family"} -->
+<h3 class="has-league-gothic-font-family" style="text-transform:uppercase">COURSE LIST</h3>
```
This string should be translated

Imran92 replied:
Translated here https://github.com/Automattic/sensei-theme/pull/44/commits/efb98493e7afa7a1576dbb5b291a0b5cd2a12328

gabrielcaires commented on patterns/homepage.php:
```diff
@@ -0,0 +1,230 @@
+<!-- /wp:group -->
+
+<!-- wp:paragraph {"className":"has-system-font-family course-font-color-tertiary","fontSize":"xx-small"} -->
+<p class="has-system-font-family course-font-color-tertiary has-xx-small-font-size"><a href="http://test.test">Explore all courses</a></p>
```
Translated string

Imran92 replied:
Translated here https://github.com/Automattic/sensei-theme/pull/44/commits/efb98493e7afa7a1576dbb5b291a0b5cd2a12328

gabrielcaires reviewed (changes requested):
I left some comments about splitting the pattern into small ones.

scruffian commented on functions.php:
```diff
@@ -68,6 +68,47 @@ function course_theme_init() {
 			'label' => __( 'Button', 'course' ),
 		)
 	);
+	register_block_style(
```
Why do we need block styles for this?

Imran92 replied:
Thanks for taking a look @scruffian, I've removed the block style [here](https://github.com/Automattic/sensei-theme/pull/44/commits/a3cde4a814d1a42e2c865715a54c7a9d9a3832f5)

MaggieCabrera commented on theme.json:
```diff
@@ -156,11 +180,21 @@
 				},
+				{
+					"name": "3.5X Large",
+					"size": "clamp(64px, calc(4rem + ((1vw - 4.8px) * 6.3889)), 110px);",
```
you might want to consider using the [built-in tools](https://make.wordpress.org/core/2022/05/27/block-font-sizes-and-fluid-typography/) for fluid typography

Imran92 replied:
I've just checked it and it looks awesome! Thanks for the great suggestion @MaggieCabrera ! As we have multiple font sizes using clamp, I'll open a separate PR changing the clamps to fluids and check to make sure nothing breaks

gabrielcaires reviewed (changes requested):
@Imran92 I think the footer is a template part, so it will update all other templates with the correct footer

donnapep commented:
Looks like the block pattern and template use "Homepage" instead of "Home Page" as per the issue.

donnapep commented:
Just to be sure I understand correctly (I think the PR description may be a bit stale), I need to add each pattern individually to the page? Is it possible to add the patterns automatically when I select the Home Page template?

Also, I don't think we need to preface each pattern with "Home Page", as it implies those patterns can only be used on a home page.

donnapep commented on patterns/blog-posts.php:
```diff
@@ -0,0 +1,39 @@
+<!-- /wp:group -->
+
+<!-- wp:paragraph {"fontSize":"xx-small","fontFamily":"system"} -->
+<p class="has-system-font-family has-xx-small-font-size"><a href="http://test.test">Visit the blog</a></p>
```
Should we just use `#` for this URL?

Imran92 replied:
Updated it [here](https://github.com/Automattic/sensei-theme/pull/44/commits/b73fbcff2137faa38f717e7d691953bc5072288d), Thanks!

donnapep commented on patterns/course-list-homepage.php:
```diff
@@ -0,0 +1,64 @@
+<!-- /wp:group -->
+
+<!-- wp:paragraph {"style":{"elements":{"link":{"color":{"text":"var:preset|color|tertiary"}}}},"fontSize":"xx-small","fontFamily":"system"} -->
+<p class="has-link-color has-system-font-family has-xx-small-font-size"><a href="http://test.test">Explore all courses</a></p>
```
I'm guessing this should probably link to the page set in Sensei > Settings > Course Archive Page.

Imran92 replied:
Updated it [here](https://github.com/Automattic/sensei-theme/pull/44/commits/d0018b883fc6a62fac73f6a30d24b3b370eac183)

donnapep commented:
Looks like the block pattern and template are named "Homepage" instead of "Home Page".  I don't think we need to prefix each pattern with "Home Page", as it implies those patterns can only be used on a home page.

The testimonial image is not working for me, as the URL appears to be hardcoded.

I find the pattern behaviour a little strange. Here's what I did:

- Created a page I called "Home".
- Set the template to Home Page.
- Added the Homepage pattern.
- Added the Course List pattern.
- Added the Newsletter pattern.
- Added the Testimonial pattern.

When I viewed the page on the frontend, I was wondering why blocks were duplicated, then realized that the Homepage pattern was already including all those other patterns, and so I didn't need to add them separately. But it was confusing because the "Homepage" pattern preview image only shows the header.

Should we consider adding the pattern automatically when I select the Home Page template instead?

I think it would be wise to pull @burtrw and @Luchadores in at this point to get their early feedback, as this will also affect the Home Page Hero PR. Ronnie and Andrei, it would be appreciated if you could test this PR a bit to see if this is the flow you expect for creating a home page. These are the patterns this PR has added:

![Screen Shot 2022-10-27 at 7 04 06 AM](https://user-images.githubusercontent.com/1190420/198268155-a2fe04fc-d6d1-43ec-968f-11e52df6bb59.jpg)

donnapep commented:
Let's not worry much about styling the footer as part of this PR, as there's a separate card for that.

donnapep commented:
It looks like the Course List pattern could use a bit more testing. When none of my courses have images, the course list has a lot of extra whitespace, and there also appears to be extra whitespace above the title. Please let me know if you think this is an issue with the Course List block itself (in which case I'll open a separate Issue), or if it's related to this PR.

The Continue button is also not styled (and I'm guessing View Results won't be either). Please see[ my PR here](https://github.com/Automattic/sensei-theme/pull/52) for addressing button styles. I hope this PR can use the styles implemented there without needing to add its own:

![Screen Shot 2022-10-27 at 7 24 53 AM](https://user-images.githubusercontent.com/1190420/198272054-4cade4f8-60ff-4c68-9d5a-1d1f75dc1c35.jpg)


The styles for this PR appear to be affecting the Course List block on other pages. We'll need to isolate any styling changes to just the home page.

## Before
![Screen Shot 2022-10-27 at 7 12 16 AM](https://user-images.githubusercontent.com/1190420/198269656-e7084e14-0f2e-416e-96ab-7b148080a131.jpg)

## After
![Screen Shot 2022-10-27 at 7 12 57 AM](https://user-images.githubusercontent.com/1190420/198269738-e697c9b4-1682-4175-9d2c-2b3eb90eb562.jpg)

gabrielcaires commented on patterns/testimonial-homepage.php:
```diff
@@ -0,0 +1,38 @@
+<!-- wp:columns {"style":{"spacing":{"blockGap":"20px"}},"textColor":"tertiary"} -->
+<div class="wp-block-columns has-tertiary-color has-text-color"><!-- wp:column {"verticalAlignment":"bottom","width":"41%"} -->
+<div class="wp-block-column is-vertically-aligned-bottom" style="flex-basis:41%"><!-- wp:image {"id":2150,"sizeSlug":"full","linkDestination":"none","style":{"border":{"radius":"8px","width":"1px"}}} -->
+<figure class="wp-block-image size-full has-custom-border"><img src="http://sentest.test/wp-content/themes/sensei-theme/assets/images/course-theme-testimonial.png" alt="" class="wp-image-2150" style="border-width:1px;border-radius:8px"/></figure>
```
Replace `http://sentest.test` with `/`

Imran92 replied:
Updated as a part of [this](https://github.com/Automattic/sensei-theme/pull/44/commits/46592c3ecfd630ed62b365a18edd4411de136d34), thanks!

gabrielcaires commented on patterns/homepage.php:
```diff
@@ -0,0 +1,36 @@
+<!-- /wp:spacer -->
+
+<!-- wp:paragraph {"align":"center","style":{"typography":{"textTransform":"uppercase","lineHeight":"1"}},"textColor":"primary","fontSize":"3-2x-large"} -->
+<p class="has-text-align-center has-primary-color has-text-color has-3-2-x-large-font-size" style="line-height:1;text-transform:uppercase">grow your writing Skills with Course</p>
```
Here should be the page title (I think`core/post-title`.)

Imran92 replied:
I'm not sure we should use the page title here, as in farther discussion, we decided to use a template with no title. The user can edit it to put any text they want here

gabrielcaires commented on patterns/footer.php:
```diff
@@ -7,6 +7,60 @@
+<!-- /wp:group -->
+
+<!-- wp:heading {"textAlign":"center","fontSize":"5x-large","fontFamily":"sorts-mill-goudy"} -->
+<h2 class="has-text-align-center has-sorts-mill-goudy-font-family has-5-x-large-font-size">COURSE</h2>
```
Here I think is the site title block

Imran92 replied:
Thanks Gabriel! I think we should take a look at it separately under footer issue ☺️

gabrielcaires reviewed (changes requested):
In addition to the Donna's comments,  I found other small points.

Luchadores commented:
Testing it right now on this Jurassic Ninja site: https://motionless-flea.jurassic.ninja/wp-admin/

- When I have no courses, in the course list block, the "No results found" is not visible enough as it's Green on Green almost. 

<img width="1032" alt="image" src="https://user-images.githubusercontent.com/10038976/198303933-9fcba553-7ca2-446a-9e40-fc69236b6f6b.png">

- Sidebar opened + switching the Course List block to the Grid View (which I would prefer to have selected by default, I got the list view by default), looks quite broken. (This is close to 1540px wide as a resolution, so there should be space to handle it better, I assume, maybe I'm wrong?).
<img width="1529" alt="image" src="https://user-images.githubusercontent.com/10038976/198309112-b902da37-a86b-450a-9bc4-4cb0ecb24791.png">
Max resolution screenshot for grid view:
<img width="2548" alt="image" src="https://user-images.githubusercontent.com/10038976/198309635-fa1c9cfd-af29-46c7-bf83-875d16fcebf3.png">

- Course List Block -> I also noticed the Start Course buttons are too wide compared to the rules we have for buttons. I would also limit the Course List body text to 3 lines of text. That should save us some vertical space.
<img width="1065" alt="image" src="https://user-images.githubusercontent.com/10038976/198312989-b5bc5890-3df3-453e-af5d-7d19b078fe91.png">


- Clicking on the "Homepage" pattern, adds the entire homepage, which is great. But it's also not what I expected since I have the other single blocks listed as options there. I just expected to add the header and add the other blocks one by one. Maybe this is part of the test :) 
<img width="332" alt="image" src="https://user-images.githubusercontent.com/10038976/198304421-8da9e160-4e3a-4073-8a88-e761f6d1bc73.png">

- When any of the sidebar menus are open, the background colors stick right to the edges of the content (Check the Course List block background to see what I mean in the screenshot below)
<img width="1529" alt="image" src="https://user-images.githubusercontent.com/10038976/198304881-fb22eda5-26db-4f4b-a23d-e020d4590fed.png">

- The testimonial block renders with a broken image
- The testimonial block spacing for the author and profession is off. Check the block in the design (we should have 10px in-between the author name and profession).
<img width="1294" alt="image" src="https://user-images.githubusercontent.com/10038976/198305082-ca6662f9-4d83-4f84-9d7d-839cfd9929f6.png">
![image](https://user-images.githubusercontent.com/10038976/198305339-edeb32a0-a740-4099-adca-d17c0a333513.png)


**Mobile**

Switching to the mobile view in the editor renders a lot of spacing problems. I will add some screenshots below of what I'm seeing.

<img width="368" alt="image" src="https://user-images.githubusercontent.com/10038976/198313764-061ab6e6-ed14-42f5-9b81-eeb7cde30fd9.png">
<img width="380" alt="image" src="https://user-images.githubusercontent.com/10038976/198313829-41339ade-3245-442a-86b1-ce4ffdf6e28c.png">
<img width="370" alt="image" src="https://user-images.githubusercontent.com/10038976/198313894-108def1b-9302-40cc-80ba-7398ce186576.png">
<img width="380" alt="image" src="https://user-images.githubusercontent.com/10038976/198313963-82f62895-ab05-4ed2-9b0e-208d8fab4c9a.png">
<img width="375" alt="image" src="https://user-images.githubusercontent.com/10038976/198314041-6e7d9085-6271-48ef-9341-bde4ae9b25d3.png">
<img width="370" alt="image" src="https://user-images.githubusercontent.com/10038976/198314082-595fc310-0b1e-46ff-b045-c98904d9dd67.png">



PS: This looks awesome to scroll through on a desktop! 👍  
PS2: Please check the [Figma file](https://www.figma.com/file/nf4oV51c1JBxJKXshMe7cd/New-Sensei-Theme?node-id=20020040%3A18973) for spacing on both Desktop and Mobile.

Imran92 commented:
Thanks for taking a look @donnapep!

> Looks like the block pattern and template are named "Homepage" instead of "Home Page". I don't think we need to prefix each pattern with "Home Page", as it implies those patterns can only be used on a home page.

Changed it to Landing page and removed prefixes [here](https://github.com/Automattic/sensei-theme/pull/44/commits/0e9b310fb1d7d7db1274caed04eb1b97928fac4f)

> The testimonial image is not working for me, as the URL appears to be hardcoded.

Fixed as a part of [this commit](https://github.com/Automattic/sensei-theme/pull/44/commits/46592c3ecfd630ed62b365a18edd4411de136d34)

> When I viewed the page on the frontend, I was wondering why blocks were duplicated, then realized that the Homepage pattern was already including all those other patterns, and so I didn't need to add them separately. But it was confusing because the "Homepage" pattern preview image only shows the header.

Hopefully our Landing Page naming approach will help with that! :)

For course list related issues, as we've discussed, hopefully course list part will get fully fixed with the button styles and course list issue is completed! We'll probably need to think about some non ideal cases like when some courses have image and some don't, or all of them not having photos and come up with a solution to make it look good in all cases! ☺️ 

Thanks so much!

Imran92 commented:
Thanks for taking a look @Luchadores ! I've updated the PR and hopefully the issues will be solved. The issues regarding the Course List will be handled in a separate PR. Please take a look when you can ☺️ And we've changed the name of the page template this time. Please along with the **Landing Page** pattern, use the **Page - No Title** template.

gabrielcaires commented on patterns/course-list.php:
```diff
@@ -0,0 +1,64 @@
+<!-- /wp:group -->
+
+<!-- wp:paragraph {"style":{"elements":{"link":{"color":{"text":"var:preset|color|tertiary"}}}},"fontSize":"xx-small","fontFamily":"system"} -->
+<p class="has-link-color has-system-font-family has-xx-small-font-size"><a href="<?php echo class_exists( 'Sensei_Course' ) ? Sensei_Course::get_courses_page_url() : '#'; ?>">Explore all courses</a></p>
```
Translate string

gabrielcaires commented on patterns/course-list.php:
```diff
@@ -0,0 +1,64 @@
+<div class="wp-block-group alignfull has-tertiary-color has-primary-background-color has-text-color has-background" style="padding-top:80px;padding-right:20px;padding-bottom:80px;padding-left:20px"><!-- wp:group {"style":{"spacing":{"padding":{"top":"0px","right":"0px","bottom":"40px","left":"0px"},"blockGap":"0"}},"backgroundColor":"primary","textColor":"tertiary","layout":{"type":"flex","flexWr ...
+<div class="wp-block-group has-tertiary-color has-primary-background-color has-text-color has-background" style="padding-top:0px;padding-right:0px;padding-bottom:40px;padding-left:0px"><!-- wp:group {"style":{"spacing":{"padding":{"left":"20px"}},"border":{"top":{"width":"0px","style":"none"},"right":{"width":"0px","style":"none"},"bottom":{"width":"0px","style":"none"},"left":{"color":"var:prese ...
+<div class="wp-block-group" style="border-top-style:none;border-top-width:0px;border-right-style:none;border-right-width:0px;border-bottom-style:none;border-bottom-width:0px;border-left-color:var(--wp--preset--color--tertiary);border-left-width:1px;padding-left:20px"><!-- wp:heading {"level":3,"style":{"typography":{"textTransform":"uppercase","lineHeight":"1"}},"fontSize":"medium","fontFamily":" ...
+<h3 class="has-league-gothic-font-family has-medium-font-size" style="line-height:1;text-transform:uppercase">COURSE LIST</h3>
```
Translate string

gabrielcaires commented:
I found some layout issues, my suggestion is to use this tool to compare the sizes, it changed my life:
https://chrome.google.com/webstore/detail/smart-page-ruler/nmibbjghlmdiafjolcphdggihcbcedmg

1) Should be 40px:
<img width="1044" alt="Screen Shot 2022-10-28 at 18 47 57" src="https://user-images.githubusercontent.com/38718/198738531-c4f28a23-125f-4660-8cf9-a6809209493f.png">

2) Should be 40px:
<img width="277" alt="Screen Shot 2022-10-28 at 18 50 18" src="https://user-images.githubusercontent.com/38718/198738820-548da464-1fb1-4fe9-bc5e-eccea42f0592.png">

3) Author name aligned by middle
<img width="446" alt="Screen Shot 2022-10-28 at 18 53 08" src="https://user-images.githubusercontent.com/38718/198739163-9d0d41c6-4c40-4534-845f-da42c1b24627.png">


4) Category color should be white + margin-top = 20px
<img width="1005" alt="Screen Shot 2022-10-28 at 18 55 20" src="https://user-images.githubusercontent.com/38718/198739425-61de86a2-9c0d-45da-b4b1-3004a48ad690.png">

5) Should be 40
![Uploading Screen Shot 2022-10-28 at 18.59.00.png…]()

6) Should be 20
<img width="354" alt="Screen Shot 2022-10-28 at 19 00 33" src="https://user-images.githubusercontent.com/38718/198739915-9121aead-e7ce-451d-9a60-77971e72769f.png">


 7) Should be 40 (left)
<img width="207" alt="Screen Shot 2022-10-28 at 19 03 06" src="https://user-images.githubusercontent.com/38718/198740160-42e60703-86c6-4d47-950a-4502cf2760bc.png">

I think that fixing it we can finally merge our cards

Imran92 commented:
Thanks for taking a look into it @gabrielcaires <3 I've addressed the spacing issues [here](https://github.com/Automattic/sensei-theme/pull/44/commits/90b07abb84bf6e2c3ee7dea8b1c37e82d8981069), and also as per our discussion in group, I've removed the course list related parts from this PR [here](https://github.com/Automattic/sensei-theme/pull/44/commits/36b0a95c776bc9ca64f00d41734b42a11be2149d), so could not address the comments regarding course list. Can you kindly take a look again? Thanks

gabrielcaires commented on patterns/newsletter-landing-page.php:
```diff
@@ -0,0 +1,22 @@
+<?php
```
is this file using the correct name? It doesn't seem like a full page.
